metal/datasource_metal_spot_market_request_test.go

```diff
@@ -21,9 +24,24 @@ func TestAccDataSourceMetalSpotMarketRequest_Basic(t *testing.T) {
 			{
 				Config: testDataSourceMetalSpotMarketRequestConfig_Basic(projectName),
 				Check: resource.ComposeTestCheckFunc(
-					testAccCheckMetalSpotMarketRequestExists("metal_spot_market_request.req", &key),
-					resource.TestCheckResourceAttr(
-						"data.metal_spot_market_request.dreq", "device_ids.#", "2"),
```

Comment:
I modified this test to not wait for devices and to offer so low a bid that it should not be accepted (to make sure the SMR can be quickly created and deleted).

The PlanOnly step ensures that a change from a facility to a Metro based SMR triggers a taint. The ID comparison in the following step ensures a new SMR was created.

I moved the data resource device count check to the resource_spot_market_request check, which is still configured to "wait" for the devices.